Makefile

```diff
@@ -12,7 +12,7 @@ else
 VERSION := $(shell git describe)
 endif
 
-LD_FLAGS = -X github.com/tendermint/tendermint/version.TMCoreSemVer=$(VERSION)
+LD_FLAGS = -X github.com/tendermint/tendermint/version.Version=$(VERSION)
```

Comment:
What's the difference between this version and the hardcoded `TMCoreSemVer = "0.34.10"` below?

Is this an override mechanism or are those different versions?

Reply:
Found my answer in the description, sorry.

But then I am consused about the two differen terms "version" and "TMCoreSemVer"

Reply:
We can change the name, not married to "version"

Reply:
"version", "TMCoreSemVer" and "software" is used to refer to the same thing.

What about "tendermint version" consistently (in whatever spelling make sense)?

Reply:
version and tmCoreSemVer can't be the same tho

Reply:
Because of the override mechanism? Should then be `tmVersion` and `tmVersionDefault` to make clear this is the fallback for the same thing?

Reply:
yup, actually doing that now.

Reply:
as for software in the version struct, we will need to do this post 0.35 as we are trying to make 0.35 non breaking

Reply:
Yeah, sure. But still good to set a target name that should be adopted eventually. You can then comment it with

```
// This is the same as `tmVersion` and should be renamed when we get the chance
```

Reply:
opening an issue and adding it to a milestone instead. I dislike todos in the code.

Reply:
ref https://github.com/tendermint/tendermint/issues/6495